components/apimgt/org.wso2.carbon.apimgt.rest.api.gateway.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/gateway/v1/impl/ApiArtifactApiServiceImpl.java

```diff
@@ -18,24 +18,101 @@
 
 package org.wso2.carbon.apimgt.rest.api.gateway.v1.impl;
 
+import org.apache.axis2.AxisFault;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
 import org.json.JSONObject;
+import org.wso2.carbon.apimgt.api.APIManagementException;
 import org.wso2.carbon.apimgt.api.gateway.GatewayAPIDTO;
+import org.wso2.carbon.apimgt.api.gateway.GatewayContentDTO;
 import org.wso2.carbon.apimgt.gateway.InMemoryAPIDeployer;
-import org.wso2.carbon.apimgt.rest.api.gateway.v1.*;
+import org.wso2.carbon.apimgt.gateway.utils.EndpointAdminServiceProxy;
+import org.wso2.carbon.apimgt.gateway.utils.LocalEntryServiceProxy;
+import org.wso2.carbon.apimgt.gateway.utils.SequenceAdminServiceProxy;
+import org.wso2.carbon.apimgt.impl.dao.ApiMgtDAO;
+import org.wso2.carbon.apimgt.rest.api.gateway.v1.ApiArtifactApiService;
 import org.apache.cxf.jaxrs.ext.MessageContext;
+import org.wso2.carbon.endpoint.EndpointAdminException;
+
 
 import javax.ws.rs.core.Response;
 
 public class ApiArtifactApiServiceImpl implements ApiArtifactApiService {
 
-    public Response apiArtifactGet(String apiName, String label, String apiId, MessageContext messageContext) {
+    private static final Log log = LogFactory.getLog(ApiArtifactApiServiceImpl.class);
+
+    private ApiMgtDAO apiMgtDAO = ApiMgtDAO.getInstance();
```

Comment:
We should not access DAO layer outside impl. Even in gateway profile, we might not have access to db.

Reply:
https://github.com/wso2/carbon-apimgt/pull/8784/commits/79c1b54ed87ee44cc8828ac3065a843208a9967e